### ISSUE TITLE
how to update a custom field value?

### ISSUE THREAD

**Issue by paulmoseley** [EncompassRest/EncompassRest#341]
I've done a lot of research/reading today, and am still trying to figure out the best way to update a custom field.  I read the closed issue(s) about this, and they seemed to indicate that I needed to add a custom field, and set a lot of things equal to themselves, in order to get the custom field value updated.  This would indicate that, if I wanted to update an existing custom field, I'd need to delete it and add it back again, which seems weird.  I noticed that, once I get a loan using GetLoanAsync, I can update custom field values if I know the index # of the custom field in the collection, like so:

**loan.CustomFields[index#].StringValue = "X";**

Will this work?  Or do I need to do other things to make sure the field update is noticed by UpdateLoanAsync?   Still working on the best way to grab the index of a custom field with a particular field name; I suppose I could read through the list/collection and find it that way (?).  I'm also wondering how often the order of items in the CustomFields collection changes, and whether I could run a service periodically which would grab and store index #'s for custom fields I care about.  Any and all thoughts appreciated...thanks......

**Comment by TylerBrinkley** [comment]
I'd recommend using the `Fields` property on `Loan` for making field updates instead of manually manipulating the loan object. https://github.com/EncompassRest/EncompassRest/wiki/Field-ID-Support

**Comment by paulmoseley** [comment]
Thanks, Tyler!  That's what I was trying to do initially, but I stubbed my toe on it somehow...maybe because I was trying things out with a loan that had been returned with a filtered LoanPipelineData collection.  I was seeing that the loan.Fields["Fields.CX.WHATEVER"] item was read-only, with a getter only, not a setter.  Or maybe it was because I neglected to add the '.Value' part on there (*blush*)...
